src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -171,6 +174,40 @@ private byte[] GetBytesFromSlave(int count, TimeInterval timeInterval)
             return result;
         }
 
+        // This method is for getting OA1EN, 
+        // returns true when OA1EN is enabled.
+        private bool OA1ENEnabled(){
+            return currentDoubleSlave.ReadDoubleWord(0x08) >> 15 != 0x00;
+        }
+
+        // This method is for getting RXEN,
+        // returns true when the buffer is empty.
+        private bool RXNECleared(){
+            return (currentDoubleSlave.ReadDoubleWord(0x18) & (1 << 2)) == 0x00;
+        }
+        
+        private delegate bool RegisterBitAccessDelegate();
+
+        private void PollForRegisterBit(RegisterBitName bitName){
+            int i = 0;
+            RegisterBitAccessDelegate registerAccess;
+            switch(bitName){
+                case RegisterBitName.OA1EN:
+                    registerAccess = OA1ENEnabled;
+                    break;
+                case RegisterBitName.RXNE:
+                    registerAccess = RXNECleared;
+                    break;
+                default:
+                    registerAccess = null;
+                    break;
+            }
+            while(!registerAccess() && i < 8){
+                Thread.Sleep(500);
```

Comment:
Is this 0.5s? Can't we wait for shorter in order to have faster flashes?

Reply:
That's correct. Since this sleep is in a loop I would also suggest reducing it to say 100ms (or even less if you find it giving better results).